corehq/apps/enterprise/api/resources.py

```diff
@@ -259,12 +242,35 @@ def dehydrate(self, bundle):
 
         return bundle
 
-    def alter_list_data_to_serialize(self, request, data):
-        result = super().alter_list_data_to_serialize(request, data)
-        path = urljoin(request.get_full_path(), 'schema/#feed')
-        result['@odata.context'] = request.build_absolute_uri(path)
+    def get_primary_key(self):
+        return 'report_name'
 
-        return result
+
+class FormSubmissionResource(ODataResource):
+    form_name = fields.CharField()
+    submitted = fields.DateTimeField()
+    app_name = fields.CharField()
+    mobile_user = fields.CharField()
+    domain = fields.CharField()
+
+    def get_object_list(self, request):
+        enddate = datetime.strptime(request.GET['enddate'], '%Y-%m-%d') if 'enddate' in request.GET else None
+        startdate = datetime.strptime(request.GET['startdate'], '%Y-%m-%d') if 'startdate' in request.GET else None
```

Comment:
Should these be dates or datetimes?  FWIW there's also `date.fromisoformat()` that might work here.  I've also found the `dateutil.parser` `parse` function useful for [this kind of thing](https://github.com/dimagi/commcare-hq/blob/1a70089d74f4c7c367799ef83e7eddcd2739340a/corehq/apps/api/util.py#L79-L105), depending on how flexible it needs to be with the input.

Reply:
From a practicality standpoint, I think they probably should remain as dates.  I think users are primarily thinking "Give me all reports between these dates (in my local timezone)". It's probably worse usability to force the user to specify a full datetime if they almost never are concerned with the time portion.  We had a call earlier today where a client mentioned they typically pull by quarter, so I think thinking about the times they'd need to specify would be confusing.

That said, typing this out points to a bug, where I'm not localizing these dates into the user's timezone

Reply:
I skipped over the second part of your suggestion. Not sure if you were suggesting using `parse_str_to_date` to support both dates and datetimes. Definitely an option, but I think I'd prefer to say simple with the API interface until we hear from our clients that they need datetime support

Reply:
@mjriley can you add tests to catch user timezone issues?

Reply:
I probably spoke too soon. The current implementation is consistent, where all dates and times are assumed to be in UTC. While I think it's likely that users will want to specify dates in their local timezone, I also think it's very likely users will be confused when they see the form's UTC timestamps looking like they are outside of the date range specified. For example, say you request 7/1 to 7/20, where your local timezone is EST. If a form was submitted at 9 PM EST on 7/20, it would be represented as 1 AM on 7/21 in UTC. I think a user would be understandably confused seeing a 7/21 date when they specified forms up to 7/20.  The most consistent report would be to localize all dates and times on the report, but then we're talking about changing the data on existing reports.

Given that the current implementation is consistent, with the only confusion coming from the user specifying new parameters, I'm leaning towards keeping it this way until we can evaluate if we want to localize everything. Do you think that makes sense, or do you think our users won't be confused by the localized dates conflicting with the UTC report values?

Reply:
In the context of an API it seems reasonable to say that all times are in UTC.  Does PowerBI not handle that on the other side of things?

> Not sure if you were suggesting using `parse_str_to_date` to support both dates and datetimes.

I wasn't, just that that's maybe a useful reference depending on needs.  That was inspired by Elasticsearch's handling of dates and datetimes, where they match filtering to the degree of precision specified - if you pass in a day, it matches anything in that day, and if you pass in a minute, it matches the whole minute.  Seems fine to do dates only here if that's what the requirements call for.

What I was commenting on was that `datetime.strptime` returns a datetime object, but it looks like you want date objects instead.  For that, `date.fromisoformat()` works well

Reply:
PowerBI does handle UTC, it's more that I think users would prefer to specify their date parameters as dates rather than full datetimes, and when they do so, they will likely believe them to be specified in their local time. If we don't think that confusion will exist, and the user will intuit that the dates they have provided as well as the report data is all going to be in UTC, no problem should exist.

As mentioned offline, while I think the users will want to pass in date parameters, on our end, we should be keeping them as full datetimes, because we don't work with naive dates. Whether we go with UTC or allow local timezones, dates have to become datetimes to support timezones.